modules/xmpp/JingleSessionPC.js

```diff
@@ -22,6 +22,148 @@ const logger = getLogger(__filename);
+        this._workFunction = workFunction;
+
+        /**
+         * The remote SDP that will be used for renegotiation. If set then
+         * then the renegotiation will be attempted regardless of the
+         * {@link shouldRenegotiate} value.
+         * @type {Object|null}
+         * @private
+         */
+        this._remoteDescription = null;
+
+        /**
+         * Stores the execution error.
+         * @type {string|null}
+         * @private
+         */
+        this._error = null;
+
+        /**
+         * Indicates whether or not renegotiation should be attempted.
+         * Both {@link _error} and {@link _remoteDescription} take precedence
+         * over this field's value.
+         * @type {boolean}
+         * @private
+         */
+        this._renegotiate = false;
+    }
+
+    /**
+     * Executed the underlying worker function by passing this instance as
+     * an argument. See class description for more info.
+     */
+    execute() {
+        this._workFunction(this);
+    }
+
+    /**
+     * Obtains the error description which occurred during worker function
+     * execution. If not <tt>null</tt> the renegotiation will not happen.
+     * @return {string|null}
+     */
+    getError() {
+        return this._error;
+    }
+
+    /**
+     * Obtains the new remote description which should be used in
+     * the renegotiation. If non <tt>null</tt> it means that the renegotiation
+     * needs to happen and there's no need to call {@link setShouldRenegotiate}.
+     * @return {Object|null}
+     */
+    getRemoteDescription() {
+        return this._remoteDescription;
+    }
+
+    /**
+     * Sets an error description which is the reason for the task execution
+     * failure. If non <tt>null</tt> the renegotiation will not happen.
+     * @param {string|null} error
+     */
+    setError(error) {
+        this._error = error;
+    }
+
+    /**
+     * Sets the new remote description which is to be used during
+     * the renegotiation. When set to not <tt>null</tt> there's no need to call
+     * {@link setShouldRenegotiate} - the renegotiation will be attempted
+     * anyway.
+     * @param {Object|null} newRemoteDesc the remote SDP instance as defined by
+     * the WebRTC.
+     */
+    setRemoteDescription(newRemoteDesc) {
+        this._remoteDescription = newRemoteDesc;
+    }
+
+    /**
+     * Sets to <tt>true</tt> during worker function execution if
+     * the renegotiation should be attempted, or <tt>false</tt> otherwise.
+     * @param {boolean} renegotiate
+     */
+    setShouldRenegotiate(renegotiate) {
+        this._renegotiate = renegotiate;
+    }
+
+    /**
+     * Indicates whether or not the renegotiation should follow worker function
+     * execution. This value is examined ony after both {@link getError} and
+     * {@link getRemoteDescription} return <tt>null</tt> values.
+     * @return {boolean}
+     */
+    shouldRenegotiate() {
+        return this._renegotiate;
```

Comment:
shouldn't this do `return this._error === null && (this._remoteDescription !== null || this._renegotiate)` ?